src/OpenvinoModel.cpp

```diff
@@ -0,0 +1,85 @@
+
+#include "OpenvinoModel.h"
+
+OpenVINOModel::OpenVINOModel()
+{
+}
+
+OpenVINOModel::~OpenVINOModel()
+{
+}
+
+bool OpenVINOModel::loadModel(const std::string& modelPath, const std::string& device)
+{
+	bool success = false;
+	try
+	{
+		ov::Core core;
+		std::shared_ptr<ov::Model> model = core.read_model(modelPath.c_str());
+		ov::preprocess::PrePostProcessor ppp(model);
+		ppp.input().preprocess().convert_layout({ 0, 3, 1, 2 });
+		model = ppp.build();
+		compiled_model = core.compile_model(model, device);
+		success = true;
+
+	}
+	catch (std::exception& e)
+	{
+		std::cout << "ERRORS";
+		std::cout << e.what();
+	}
+	return success;
+}
+
+
+ov::Tensor OpenVINOModel::predict(const ov::Tensor& input)
+{
+	ov::InferRequest ireq = this->compiled_model.create_infer_request();
+	ireq.set_input_tensor(input);
+	ireq.infer();
+	const ov::Tensor& output_tensor = ireq.get_output_tensor();
+	return output_tensor;
+	
+}
+
+
+ov::Tensor OpenVINOModel::convertToTensor(const cv::Mat& img, bool normalize, bool color)
+{
+	cv::Mat c = img.clone();
+	if (color)
+	{
+		cv::cvtColor(c, c, cv::COLOR_BGR2RGB);
+	}
+
+	float scale = (normalize) ? 1.0 / 255.0 : 1.0;
+	int channels = c.channels();
+	auto colorRead = (channels == 3) ? CV_32FC3 : CV_32FC1;
+	c.convertTo(c, colorRead, scale);
+
+	ov::Shape input_shape = { 1, size_t(c.rows), size_t(c.cols), size_t(channels) };
+	ov::element::Type input_type = ov::element::f32;
+	ov::Tensor converted = ov::Tensor(input_type, input_shape, c.data);
```

Comment:
We don't need to create 2 tensors to copy data

```suggestion
	ov::Tensor res = ov::Tensor(input_type, input_shape);
	size_t img_data_size = img.total() * img.elemSize();
	std::memcpy(res.data(), img.data, img_data_size);
```